lib/api/core/models/repositories/repository.js

```diff
@@ -153,7 +166,12 @@ Repository.prototype.search = function (filter, from, size, hydrate) {
       promises = [];
 
       if (!hydrate || !response.data || !response.data.body.hits) {
-        return deferred.resolve(response);
+        return deferred.resolve(response.data.body.hits.map((document) => {
```

Comment:
Same here: if this block statement is executed because `response.data` does not exist, then accessing to `response.data.body.hits` might generate an exception.